cpp/ql/src/semmle/code/cpp/ir/implementation/unaliased_ssa/Instruction.qll

```diff
@@ -1831,8 +1831,29 @@ class UnreachedInstruction extends Instruction {
  * An instruction representing a built-in operation. This is used to represent
  * operations such as access to variable argument lists.
  */
-class BuiltInInstruction extends Instruction {
+class BuiltInOperationInstruction extends Instruction {
+  Language::BuiltInOperation operation;
+
+  BuiltInOperationInstruction() {
+    getOpcode() instanceof BuiltInOperationOpcode and
+    operation = Construction::getInstructionBuiltInOperation(this)
+  }
+
+  final Language::BuiltInOperation getBuiltInOperation() {
+    result = operation
+  }
+}
+
+/**
+ * An instruction representing a built-in operation that does not have a specific opcode. The
+ * actual operation is specified by the `getBuiltInOperation()` predicate.
+ */
+class BuiltInInstruction extends BuiltInOperationInstruction {
   BuiltInInstruction() {
-    getOpcode() instanceof BuiltInOpcode
+    getOpcode() instanceof Opcode::BuiltIn
```

Comment:
Is there now no `Instruction` class for the varargs builtins?

Reply:
What used to be called `BuiltInInstruction`, which was a base class for the varargs instructions, is now called `BuiltInOperationInstruction`. What is now called `BuiltInInstruction` is the class that specifically represents instructions with the opcode `BuiltIn`, which does not include the varargs instructions, which still have their own opcode.